### ISSUE TITLE
Rule SC2034 not completely catching read -i "y" situations

### ISSUE THREAD

**Issue by MrScrith** [koalaman/shellcheck#1029]
#### For bugs
- Rule Id: SC2034 
- My shellcheck version: 0.4.6 
- I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

read -e -r -p "would you like to save? " -i "y" SAV
read -e -r -p "what version would you like to save?" -i "2" VER

echo "$SAV - $VER"
```

#### Here's what shellcheck currently says:
Line 3:
read -e -r -p "would you like to save? " -i "y" SAV
                                            ^-- SC2034: y appears unused. Verify it or export it.

#### Here's what I wanted or expected to see:

No issues.

Note that line 4 ( -i "2" VER) isn't a problem, shellcheck is properly catching when it is a number but not when it is an alphabet character.

**Comment by floatingatoll** [comment]
I also observed this false positive with multi-character `-i` values, and did not observe it with multi-character numeric `-i` values:

```
read -e -p "Stringic: " -i "STRING" VARNAME
read -e -p "Numeric: " -i "1234" VARNAME
```
